src/components/Login/Login.jsx

```diff
@@ -4,46 +4,63 @@ import { Redirect } from 'react-router-dom';
 import classNames from 'classnames/bind';
+import github from '../../images/github.png';
 
 const cx = classNames.bind(styles);
 
 export class Login extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
-            value: ""
+            username: "",
+            password: "",
+            signUp: true
         };
-        this.sendLogin = this.sendLogin.bind(this);
-        this.handleChange = this.handleChange.bind(this);
+        this.sendLoginOrSignUp = this.sendLoginOrSignUp.bind(this);
+        this.handleChangeUsername = this.handleChangeUsername.bind(this);
+        this.handleChangePassword = this.handleChangePassword.bind(this);
+        this.toggleSignUp = this.toggleSignUp.bind(this);
     }
 
     componentDidMount() {
         this.props.authenticateUser();
     }
 
-    handleChange(event) {
-        this.setState({ value: event.target.value });
+    handleChangeUsername(event) {
+        this.setState({ username: event.target.value });
     }
 
-    sendLogin(event) {
+    handleChangePassword(event) {
+        this.setState({ password: event.target.value });
+    }
+
+    sendLoginOrSignUp(event) {
         event.preventDefault();
-        this.props.logIn(this.state.value);
+        console.log(this.state.username, this.state.password);
+           
+        (this.state.signUp) ? this.props.signUp(this.state.username, this.state.password) : this.props.logIn(this.state.username, this.state.password);
+    }
+
+    toggleSignUp() {
+        this.setState({ signUp: !this.state.signUp });
     }
 
     render() {
         return (
             (!this.props.username) ?
                 <div className={styles.loginPage}>
                     <Logo />
-                    <form className={styles.loginForm} onSubmit={this.sendLogin} data-cy='login-form' >
-                        <h2 className={styles.formHeader}>Type a username to enter the game:</h2>
-                        <input data-cy='username' className={cx(styles.loginInput, { inputError: this.props.error })} value={this.state.value} placeholder="Enter username" onChange={this.handleChange} autoFocus />
+                    <form className={styles.loginForm} onSubmit={this.sendLoginOrSignUp} data-cy='login-form' >
+                        <h2 className={styles.formHeader}>{this.state.signUp ? "Enter a username and password to sign up for the game" : "Log in with your username and password to enter the game"}</h2>
+                        <input type='text' data-cy='username' className={cx(styles.loginInput, { inputError: this.props.error })} username={this.state.username} placeholder="Enter username" onChange={this.handleChangeUsername} autoFocus />
+                        <input type='password' data-cy='password' className={cx(styles.loginInput, { inputError: this.props.error })} password={this.state.password} placeholder="Enter password" onChange={this.handleChangePassword} autoFocus />
```

Comment:
there was a famous password leak that occurred because React was managing the data inside a `<input type="password"`. I've been unable to find the article, but I wonder whether it's safe to bind `password` to `this.state`. I think to secure it, you need to ensure that javascript doesn't touch sensitive data.

Reply:
ordinarily, I think javascript cannot see the value inside a `<input type="password">`, but this model-view binding circumvents the browser's security.

Reply:
How would you do it instead?